source/common.ts

```diff
@@ -81,22 +78,18 @@ const i18n = {
     return Promise.resolve();
   },
   _locale: 'en',
-  _localeData(locale?: string) {
-    locale = i18n.normalize(locale ?? i18n.getLocale());
-    return locale && i18n._locales[locale.toLowerCase()];
-  },
-  _locales: LOCALES,
   _logger(error: unknown) {
     console.error(error);
   },
   _normalizeWithAncestors(locale = '') {
+    
     if (!(locale in i18n._normalizeWithAncestorsCache)) {
       const locales: string[] = [];
-      const parts = locale.toLowerCase().split(/[-_]/);
+      const parts = locale.split(/[-_]/);
       while (parts.length) {
         const locale = parts.join('-');
-        if (locale in i18n._locales) {
-          locales.push(i18n._locales[locale][0]);
+        if (locale.match(/^[a-z]{2}$|^[a-z]{2}-[A-Z]{2}$/g)) {
```

Comment:
1. This regex should be extracted into a constant.
    * We could store it on the `i18n` object so the end-user could override it. For example, one may want to accept only `xx-xx` or `xx` locales.
2. I don't like the idea of it being case-sensitive. Is there any reason for that?
3. It can be simplified to `/^[a-z]{2}(?:[A-Z]{2})$/` (note that there's no `g` flag).